src/data/UserD2Repository.ts

```diff
@@ -214,9 +214,13 @@ export class UserD2Repository implements UserRepository {
             })
             .getData();
 
-        const userGroup = response.userGroups[0];
-        if (userGroup) {
-            const ids = userGroup.users.map(user => user.id);
+        const allUsers = _(response.userGroups)
+            .map(ug => ug.users)
+            .flatten()
+            .value();
+
+        if (allUsers.length > 0) {
```

Comment:
[subjective] Typically, the "guard style" is `if (allUsers.length === 0) return []`